plugin/pkg/admission/storage/persistentvolume/label/admission.go

```diff
@@ -153,16 +170,44 @@ func (l *persistentVolumeLabel) Admit(ctx context.Context, a admission.Attribute
 			// Need at least one term pre-allocated whose MatchExpressions can be appended to
 			volume.Spec.NodeAffinity.Required.NodeSelectorTerms = make([]api.NodeSelectorTerm, 1)
 		}
-		if nodeSelectorRequirementKeysExistInNodeSelectorTerms(requirements, volume.Spec.NodeAffinity.Required.NodeSelectorTerms) {
-			klog.V(4).Infof("NodeSelectorRequirements for cloud labels %v conflict with existing NodeAffinity %v. Skipping addition of NodeSelectorRequirements for cloud labels.",
-				requirements, volume.Spec.NodeAffinity)
-		} else {
-			for _, req := range requirements {
-				for i := range volume.Spec.NodeAffinity.Required.NodeSelectorTerms {
-					volume.Spec.NodeAffinity.Required.NodeSelectorTerms[i].MatchExpressions = append(volume.Spec.NodeAffinity.Required.NodeSelectorTerms[i].MatchExpressions, req)
-				}
+
+		deprecatedNodeTerms := volume.DeepCopy().Spec.NodeAffinity.Required.NodeSelectorTerms
```

Comment:
Changes here got complicated quickly. Mainly because the existing PV admission label implementation appends selector requirements to all existing node selector terms. I would not be opposed to either:
* removing the PV admission controller (this has been deprecated anyways for a long time)
* not supporting the stable labels here